src/components/alerting.tsx

```diff
@@ -726,6 +726,13 @@ const AlertsDetailsPage_: React.FC<AlertsDetailsPageProps> = ({ history, match }
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const labels: PrometheusLabels = React.useMemo(() => alert?.labels, [labelsMemoKey]);
 
+  const ruleLabelsMemoKey = JSON.stringify(rule?.labels);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const ruleLabels: PrometheusLabels = React.useMemo(() => rule?.labels, [ruleLabelsMemoKey]);
```

Comment:
Unlike with `labels`, I don't think there's really a benefit in memoizing `ruleLabels`. Memoizing `labels` prevents unnecessary rerendering of several components including the `Graph`, which is quite slow to render. Memoizing `ruleLabels` doesn't look like it will prevent any rerenders or improve performance generally.